src/services/ImageRepository.h

```diff
@@ -20,37 +20,65 @@ enum class ImageType
 class ImageReference
 {
 public:
-    ImageReference() {}
-    ImageReference(ImageType nType, const std::string& sName)
+    ImageReference() noexcept = default;
+    explicit ImageReference(ImageType nType, const std::string& sName) noexcept
     {
         ChangeReference(nType, sName);
     }
 
-    ImageReference(const ImageReference& source)
-        : m_nType(source.m_nType), m_sName(source.m_sName)
+    ImageReference(const ImageReference& source) noexcept :
+        m_nType{ source.m_nType }, m_sName{ source.m_sName }
     {
         // don't copy m_hBitmap, it'll get initialized when GetHBitmap is called, which
         // will ensure the reference count is accurate.
     }
 
-    ~ImageReference();
-    
-    /// <summary>
-    /// Updates the referenced image.
-    /// </summary>
+    ImageReference& operator=(const ImageReference&) noexcept = delete;
```

Comment:
Wasn't really sure if we needed copy assign because of the comment regarding not copying `m_hBitmap`. Code still compiles and images still show.